dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/AbstractConfigAccessorBuilder.java

```diff
@@ -0,0 +1,228 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config14.impl;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.config.ConfigAccessorBuilder;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.Converter;
+
+import com.ibm.ws.microprofile.config14.interfaces.Config14Constants;
+import com.ibm.ws.microprofile.config14.interfaces.WebSphereConfig14;
+
+public abstract class AbstractConfigAccessorBuilder<T> implements ConfigAccessorBuilder<T> {
+
+    private final WebSphereConfig14 config;
+    private final String propertyName;
+    private final List<String> suffixValues = new ArrayList<String>();
+
+    //access to these variables must be synchronized to prevent them being changed during a call to build()
+    private Duration cacheFor = Duration.ZERO;
```

Comment:
Not sure the ConfigAccessorBuilder implementation needs to be threadsafe. The interface makes no mention of it.

Reply:
Why would I not make it threadsafe? That's just asking for trouble.